namedtensor.sty

```diff
@@ -12,3 +12,19 @@
 \newcommand{\nsum}[1]{\sum\limits_{\name{#1}}}
 \newcommand{\nfun}[2]{\mathop{\mathop{\mathrm{#2}}\limits_{\name{#1}}}}
 \newcommand{\nmov}[3]{\left[#3\right]_{\name{#1}\rightarrow\name{#2}}}
+
+
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+% Boaz's commands
+% define a new axis, right now if we do \newaxis{ax} then \ax becomes a command for \name{ax}
+% when I get the time, I want to add an optional argument that allows to use a different name
+% for the command and the displayed name (in case there is a command we don't want to clobber)
+\newcommand{\newaxis}[1]{\expandafter\def \csname #1\endcsname{\name{#1}}}
+
+
+% commands dropping the \name - useful for annotated axis or more than one axis
+\newcommand{\abin}[2]{\mathbin{\mathop{#2}\limits_{#1}}} 
```

Comment:
Currently you can write \ndot{foo,bar} and you get two sans-serif names separated by a math comma (good).

You can write \ndot{foo^a} and you get a superscript sans-serif a, which I think is good.

\ndot{foo^1} gives you a sans-serif 1, which I think might be ok.

The main thing you can't get is \ndot{foo_i} where i is a variable, not a literal letter i.

Is there another way we can get this without defining a whole new series of macros? If nothing else, I'd prefer \ndot* instead of \adot.

Reply:
Since I like having an \axis macro for using axes, which I think makes LaTeX easier to read and write, it's more natural for me to use \ndot{\axis} than \ndot{axis}. I didn't check whether \ndot{\axis} works under the current definition of \ndot- maybe it does and the underset argument becomes
\ensuremath{\mathsf{\ensuremath{\mathsf{axis}}}}
Which may give out the same result - not sure.